src/modules/interfaces.ts

```diff
@@ -397,6 +401,162 @@ export class RpcBase {
+              raise exception 'Error: %', _content;
+            end if;
+          end if;
+          return query select
+            ${
+              rpcOutputEntries.length
+                ? `${rpcOutputEntries
+                    .map(([col, typ]) => `try_cast(j.s->>'${col}', NULL::${typ}) as ${col}`)
+                    .join(', ')}`
+                : ''
+            }
+          from (
+            select json_array_elements(_content::json) as s
+          ) as j;
+        end;
+        $$;
+      `;
+    const beforeUninstallSql = `
+        DROP FUNCTION "${snakeCase(key)}";
+      `;
+    return { beforeUninstallSql, afterInstallSql };
+  }
+}
+
+export class AwsSdkInvoker extends RpcBase {
+  preTransactionCheck = PreTransactionCheck.NO_CHECK;
+  postTransactionCheck = PostTransactionCheck.NO_CHECK;
+
+  clientType: keyof AWS;
+
+  outputTable = {
+    result: 'json',
+  } as const;
+
+  getInstallUninstallSql(key: string) {
+    const rpcOutputEntries = Object.entries(this.outputTable ?? {});
+    const rpcOutputTable = rpcOutputEntries
+      .map(([columnName, columnType]) => `${columnName} ${columnType}`)
+      .join(', ');
+    const afterInstallSql = `
+        create or replace function ${snakeCase(
+          key,
+        )}(method_name varchar, method_input json, region varchar) returns table (
+          ${rpcOutputTable}
+        )
+        language plpgsql security definer
+        as $$
+        declare
+          _content text;
+        begin
+          perform http.http_set_curlopt('CURLOPT_TIMEOUT_MS', '3600000');
+          select content into _content
+          from http.http_post(
+            'http://${config.http.host}:8088/v1/db/rpc',
+            json_build_object(
+              'dbId', current_database(),
+              'dbUser', SESSION_USER,
+              'params', (SELECT array[method_name, method_input::varchar, region]),
+              'modulename', '${this.module.name}',
+              'methodname', '${key}',
+              'preTransaction', '${this.preTransactionCheck}',
+              'postTransaction', '${this.postTransactionCheck}'
+            )::varchar,
+            'application/json'
+          );
+          if json_typeof(_content::json) <> 'array' then
+            if json_typeof(_content::json) = 'object' then
+              raise exception 'Error: %', _content::json->>'message';
+            else
+              raise exception 'Error: %', _content;
+            end if;
+          end if;
+          return query select
+            ${
+              rpcOutputEntries.length
+                ? `${rpcOutputEntries
+                    .map(([col, typ]) => `try_cast(j.s->>'${col}', NULL::${typ}) as ${col}`)
+                    .join(', ')}`
+                : ''
+            }
+          from (
+            select json_array_elements(_content::json) as s
+          ) as j;
+        end;
+        $$;
+      `;
+    const beforeUninstallSql = `
+        DROP FUNCTION "${snakeCase(key)}";
```

Comment:
If you do the function overloading suggested above, this now needs to be two `DROP FUNCTION` calls with the arg types following it, eg:

```suggestion
        DROP FUNCTION "${snakeCase(key)}"(varchar, json);
        DROP FUNCTION "${snakeCase(key)}"(varchar, json, varchar);
```